src/Components/Artwork/SaveButton/SaveArtworkToListsButton.tsx

```diff
@@ -0,0 +1,102 @@
+import { AuthContextModule } from "@artsy/cohesion"
+import { useToasts } from "@artsy/palette"
+import { useManageArtworkForSavesContext } from "Components/Artwork/ManageArtworkForSaves"
+import { SaveButtonBase } from "Components/Artwork/SaveButton/SaveButton"
+import { useSaveArtwork } from "Components/Artwork/SaveButton/useSaveArtwork"
+import { FC } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useTracking } from "react-tracking"
+import { SaveArtworkToListsButton_artwork$data } from "__generated__/SaveArtworkToListsButton_artwork.graphql"
+
+interface SaveArtworkToListsButtonProps {
+  artwork: SaveArtworkToListsButton_artwork$data
+  contextModule: AuthContextModule
+}
+
+const SaveArtworkToListsButton: FC<SaveArtworkToListsButtonProps> = ({
+  artwork,
+  contextModule,
+}) => {
+  const tracking = useTracking()
+  const { sendToast } = useToasts()
+  const {
+    setArtworkId,
+    savedListId,
+    isSavedToList,
+  } = useManageArtworkForSavesContext()
+
+  const isSaved = !!artwork.is_saved
+
+  const openManageArtworkForSavesModal = () => {
+    setArtworkId(artwork.internalID)
+  }
+
+  const showToastByAction = (action: string) => {
+    if (action === "Saved Artwork") {
+      sendToast({
+        variant: "success",
+        message: "Artwork saved",
+        action: {
+          label: "Add to a List",
+          onClick: openManageArtworkForSavesModal,
+        },
+      })
+
+      return
+    }
+
+    sendToast({
+      variant: "message",
+      message: "Removed from All Saves",
+    })
+  }
+
+  const { handleSave } = useSaveArtwork({
+    isSaved,
+    artwork,
+    contextModule,
+    onSave: ({ action, artwork }) => {
+      tracking.trackEvent({
+        action,
+        entity_slug: artwork.slug,
+        entity_id: artwork.internalID,
+      })
+
+      showToastByAction(action)
+    },
+  })
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    event.preventDefault()
+
+    if (savedListId) {
+      openManageArtworkForSavesModal()
+      return
+    }
+
+    handleSave()
+  }
+
+  return (
+    <SaveButtonBase
+      isSaved={savedListId ? isSavedToList : isSaved}
+      onClick={handleClick}
+    />
+  )
+}
+
+export const SaveArtworkToListsButtonFragmentContainer = createFragmentContainer(
+  SaveArtworkToListsButton,
+  {
+    artwork: graphql`
+      fragment SaveArtworkToListsButton_artwork on Artwork {
+        id
+        internalID
+        is_saved: isSaved
```

Comment:
I think we can change that to:
```suggestion
        isSaved
```

Reply:
if you don't mind, then I think it's better to do it in a separate PR and replace it in all places at once (e.g [here](https://github.com/artsy/force/blob/0177ef9df54e94bf39c8b084bc70320c903cd26a/src/Components/Artwork/FlatGridItem.tsx#L154), [here](https://github.com/artsy/force/blob/0177ef9df54e94bf39c8b084bc70320c903cd26a/src/Components/Artwork/SaveButton/SaveArtworkMutation.ts#L22), [here](https://github.com/artsy/force/blob/0177ef9df54e94bf39c8b084bc70320c903cd26a/src/Components/Artwork/SaveButton/SaveButton.tsx#L108), etc)

Reply:
although we can add these changes for this component now (I was afraid that we might break something somewhere. but after investigating the code a bit, I don't see any problems that we may encounter if we use `Issued` in the query)